src/include/storage/sinval.h

```diff
@@ -80,6 +80,14 @@ typedef struct
 	int8		id;				/* type field --- must be first */
 	Oid			dbId;			/* database ID, or 0 if a shared relation */
 	Oid			relId;			/* relation ID, or 0 if whole relcache */
+	/* 
+	 * We need a way to mark whether this inval message is for a local ENR entry,
+	 * so that we can skip inserting it into the SI message queue.
+	 * Note that checking whether the relId is an ENR entry at the time of insertion
+	 * is insufficient, as during drops the ENR entry is deleted before we perform
+	 * SI queue insertion.
+	 */
+	bool		local_only;
```

Comment:
can we do the same for catcache invalidations?

Reply:
I don't see why not. However, as it is, there don't seem to be any issues related to catcache invalidations being sent to the SI queue, so it doesn't seem to be necessary.

Reply:
adding this local_only field to SI messages essentially addresses the cache inval change as well, so we should combine the two. if the intent is also have this SI fix as part of the lock changes then catcache invalidations need to be addressed too since temp tables certainly emit catcache invals.

Reply:
then maybe we could move the cache inval changes out of this PR? i think it's best to not mix lock changes with an incomplete fix for cache inval in one PR, especially since the relcache portion is just a few lines of code.

Reply:
The thing is that this change is required to fix the issues with `relation deleted while still in use` and `relation missing (x) attributes`. Unless we still don't plan on enabling temp oids in Uakari at all, I think that this change is important for overall stability, while the changes for cache inval don't seem as necessary.

Reply:
ok. i will make the catcache changes in a separate change (#411 ) since it is a bit more invasive.